hazelcast/src/hazelcast/client/connection/InSelector.cpp

```diff
@@ -66,7 +66,7 @@ namespace hazelcast {
                             sleepingSocket->receive(&wakeUpSignal, sizeof(int));
                         } else {
                             boost::shared_ptr<Connection> conn = connectionManager.getConnectionIfAvailable(fd);
-                            if (conn.get() != NULL) {
+                            if (conn.get() != NULL && conn->live) {
```

Comment:
I don't think another wakeup call is necessary. 
`select` call already handles the socket close. 
`EBADF` is checked explicitly for that in our code.